provider/workspace_preset_test.go

```diff
@@ -265,6 +265,249 @@ func TestWorkspacePreset(t *testing.T) {
 			}`,
 			ExpectError: regexp.MustCompile("An argument named \"invalid_argument\" is not expected here."),
 		},
+		{
+			Name: "Prebuilds is set with an empty autoscaling field",
+			Config: `
+			data "coder_workspace_preset" "preset_1" {
+				name = "preset_1"
+				prebuilds {
+					instances = 1
+					autoscaling {}
+				}
+			}`,
+			ExpectError: regexp.MustCompile(`The argument "[^"]+" is required, but no definition was found.`),
```

Comment:
Nit: let's check for a precise name to ensure we don't get false positives.
Or is it like this because we can't know which argument will show up first?